src/debugger/appWorker.ts

```diff
@@ -203,6 +206,9 @@ export class MultipleLifetimesAppWorker {
             if (object.method === "prepareJSRuntime") {
                 // The MultipleLifetimesAppWorker will handle prepareJSRuntime aka create new lifetime
                 this.gotPrepareJSRuntime(object);
+            } else if (object.method === "$disconnected") {
+                // We need to shutdown the current app worker, and create a new lifetime
+                this.singleLifetimeWorker = null;
```

Comment:
Can we run into memory problems if we don't dispose of the context? How many times is this expected to happen for one debugging session? Could we maybe use `cluster` to host each sandbox in a new process like this? http://stackoverflow.com/questions/11652530/node-js-vm-how-to-cancel-script-runinnewcontext/11713088#11713088